server/src/Core.hs

```diff
@@ -124,6 +128,9 @@ processOps core = go Null
           -- temporary file will thus not corrupt the datastore
           writeFile tempFileName (encodeToLazyText newValue)
           renameFile tempFileName fileName
+          forM_ (coreMetrics core) $ \m -> do
+            stat <- Posix.getFileStatus fileName
+            Metrics.setDataSize (Posix.fileSize stat) m
```

Comment:
Maybe extract this into a separate function?

Reply:
Since this is only used once, making it a top level function seems to introduce quite a lot of overhead (because all the local variables would need to be passed as arguments). I'd rather leave it as it is.

Reply:
Ok. The argument "this is only used once" is no argument since most functions are only used once and that would be no excuse to have the whole program _in_ `main` ;) I see what you get however, altough small reasonable points like that make programs grow back hairs.